src/scopmath/sparse_thread.c

```diff
@@ -822,9 +816,7 @@ static SparseObj* create_sparseobj() {
 	SparseObj* so;
 
 	so = emalloc(sizeof(SparseObj));
-	nrn_malloc_lock();
 	so->elmpool = nrn_pool_create(100, sizeof(Elm));
-	nrn_malloc_unlock();
```

Comment:
to clarify about the removal of lock/unlock - I haven't looked whole code workflow but I "assume" this was to protect allocation/reallocation of SparseObj when threads are active. So this is no longer needed?

Reply:
My understanding is that this was intended to support systems where `malloc`, `free` etc. were not thread-safe. Both C11 and C++11 specify that they are thread-safe, so I think this can be dropped.

Reply:
> Both C11 and C++11 specify that they are thread-safe, so I think this can be dropped

Thanks for pointing that out! 👍